src/basic/errno-util.h

```diff
@@ -108,8 +114,8 @@ static inline bool ERRNO_IS_TRANSIENT(int r) {
  *
  * Hint #3: When asynchronous connect() on TCP fails because the host never acknowledges a single packet,
  *          kernel tells us that with ETIMEDOUT, see tcp(7). */
-static inline bool ERRNO_IS_DISCONNECT(int r) {
-        return IN_SET(abs(r),
+static inline bool ERRNO_IS_NEG_DISCONNECT(int r) {
+        return IN_SET(-r,
```

Comment:
So everybody is on the same page that INT_MIN is not a valid error value?

Reply:
This raises a good point. No matter what the type is used, INT_MIN = - INT_MAX - 1. So I can't do `-r` without risking overflow. I'll change the macros to keep `r` as it was, but add minus signs on the constants. That way, after the widening, `ERRNO_IS_NEG_DISCONNECT(MAXINT_MIN)` works as expected.

Reply:
I added the assert as requested.